crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -346,6 +424,27 @@ impl Schedule {
         self
     }
 
+    /// Call function `f` on each pair of ([`NodeId`], [`ScheduleSystem`]).
+    pub fn systems_for_each(&self, mut f: impl FnMut(NodeId, &ScheduleSystem)) {
```

Comment:
Does this need to be public? I think you wrote this as a helper function for the diagnostic system right?